soh/soh/Enhancements/sfx-editor/SfxEditor.cpp

```diff
@@ -0,0 +1,359 @@
+    {NA_SE_SY_START_SHOT,          {"Gunshot",                             "NA_SE_SY_START_SHOT",            SEQ_SFX}},
+    {NA_SE_IT_HAMMER_HIT,          {"Hammer Bonk",                         "NA_SE_IT_HAMMER_HIT",            SEQ_SFX}},
+    {NA_SE_EN_IRONNACK_SWING_AXE,  {"Iron Knuckle",                        "NA_SE_EN_IRONNACK_SWING_AXE",    SEQ_SFX}},
+    {NA_SE_EN_FANTOM_ST_LAUGH,     {"Phantom Ganon Laugh",                 "NA_SE_EN_FANTOM_ST_LAUGH",       SEQ_SFX}},
+    {NA_SE_EV_PLANT_BROKEN,        {"Plant Explode",                       "NA_SE_EV_PLANT_BROKEN",          SEQ_SFX}},
+    {NA_SE_SY_GET_RUPY,            {"Rupee",                               "NA_SE_SY_GET_RUPY",              SEQ_SFX}},
+    {NA_SE_VO_RT_CRASH,            {"Ruto Crash",                          "NA_SE_VO_RT_CRASH",              SEQ_SFX}},
+    {NA_SE_VO_RT_LIFT,             {"Ruto Lift",                           "NA_SE_VO_RT_LIFT",               SEQ_SFX}},
+    {NA_SE_VO_RT_THROW,            {"Ruto Thrown",                         "NA_SE_VO_RT_THROW",              SEQ_SFX}},
+    {NA_SE_EN_NUTS_UP,             {"Scrub Emerge",                        "NA_SE_EN_NUTS_UP",               SEQ_SFX}},
+    {NA_SE_EN_AWA_BOUND,           {"Shabom Bounce",                       "NA_SE_EN_AWA_BOUND",             SEQ_SFX}},
+    {NA_SE_EN_SHELL_MOUTH,         {"Shellblade",                          "NA_SE_EN_SHELL_MOUTH",           SEQ_SFX}},
+    {NA_SE_EN_STALTU_DAMAGE,       {"Skulltula Damage",                    "NA_SE_EN_STALTU_DAMAGE",         SEQ_SFX}},
+    {NA_SE_EN_NUTS_THROW,          {"Spit Nut",                            "NA_SE_EN_NUTS_THROW",            SEQ_SFX}},
+    {NA_SE_VO_TA_CRY_0,            {"Talon Hmm",                           "NA_SE_VO_TA_CRY_0",              SEQ_SFX}},
+    {NA_SE_VO_TA_SURPRISE,         {"Talon Surprised",                     "NA_SE_VO_TA_SURPRISE",           SEQ_SFX}},
+    {NA_SE_SY_LOCK_ON,             {"Target Enemy",                        "NA_SE_SY_LOCK_ON",               SEQ_SFX}},
+    {NA_SE_SY_LOCK_ON_HUMAN,       {"Target Neutral",                      "NA_SE_SY_LOCK_ON_HUMAN",         SEQ_SFX}},
+    {NA_SE_SY_FSEL_CURSOR,         {"File Select Cursor",                  "NA_SE_SY_FSEL_CURSOR",           SEQ_SFX}},
+    {NA_SE_SY_FSEL_DECIDE_L,       {"File Select Choose",                  "NA_SE_SY_FSEL_DECIDE_L",         SEQ_SFX}},
+    {NA_SE_SY_FSEL_CLOSE,          {"File Select Back",                    "NA_SE_SY_FSEL_CLOSE",            SEQ_SFX}},
+    {NA_SE_IT_BOMB_EXPLOSION,      {"Bomb Explosion",                      "NA_SE_IT_BOMB_EXPLOSION",        SEQ_SFX}},
+    {NA_SE_EV_CHICKEN_CRY_A,       {"Chicken Cry",                         "NA_SE_EV_CHICKEN_CRY_A",         SEQ_SFX}},
+};
+
+void Draw_SfxTab(const std::string& tabKey, const std::map<u16, std::tuple<std::string, std::string, SeqType>>& map, SeqType type) {
+    const std::string hiddenTabKey = "##" + tabKey;
```

Comment:
it's not quite clear what `tabKey` and `hiddenTabKey` mean when reading through this. i know it has something to do with imgui elements and putting `##` in there makes it so we don't see that part of the string

i'm not sure what the values in tabKey are supposed to represent, if i had a better idea of that i might have some suggestions for better names

Reply:
Yeah you’ve mostly got the idea, the tabkey is passed in near the bottom when we draw each tab. Maybe tabID would be better